bindings/c/test/mako/mako.c

```diff
@@ -111,6 +114,19 @@ void update_op_lat_stats(struct timespec* start, struct timespec* end, int op, m
 	if (latencyus > stats->latency_us_max[op]) {
 		stats->latency_us_max[op] = latencyus;
 	}
+	if (!is_memory_allocated[op]) return;
+	if (elem_size[op] < stats->latency_samples[op]) {
+		elem_size[op] = elem_size[op] + LAT_BLOCK_SIZE;
+		lat_block_t* temp_block = (lat_block_t*)malloc(sizeof(lat_block_t));
```

Comment:
Ah, in fact, if we wait until `malloc()` to fail, we'll have a problem when we write out the stats to `/dev/shm`.  We're pretty much out of memory at this point.  This probably won't be a big deal in practice, we still should handle this gracefully.  One safety net we can implement is to parameterize the max size of this linked list buffer.